### PR TITLE
fix: don't error when master branch is already up-to-date

### PR DESCRIPTION
**Pull request by RodEsp**
* This PR allows Leif to correctly handle the case when a branch it would create already exists but the main branch already contains all its commits so creating a PR is pointless.
Before this PR leif would throw when the GH api returned a 422 error saying there are no commits between the main branch and leif's "sync" branch.

New message looks like this:
image

### CODE REVIEW

File: src/commands/run.ts [hunk 1]
```diff
@@ -63,7 +63,7 @@ export default class Run extends Command {
         preparedWorkflows = preparedWorkflows.filter(workflow => {
           const shouldInclude = workflows.includes(workflow.id)
           if (shouldInclude) {
-            workflow.sequences = workflow.sequences.filter(sequence => sequences.includes(sequence.id))
+            workflow.sequences = workflow.sequences.filter(sequence => sequences ? sequences.includes(sequence.id) : sequence)
```
Comment (peternhale) [inline src/commands/run.ts:66]: +1 for the need for pairwise testing. The interaction of flags.workflow and flags.sequence.

**Review by peternhale — APPROVED** [review]

**Comment by RasPhilCo** [comment]
@RodEsp curious, why doesn't this diff check fail to exit? https://github.com/RasPhilCo/leif/blob/297aaf855826463c4beb665d37fdf310eb71277a/src/sequences.ts#L94-L112

**Comment by RodEsp** [comment]
@RasPhilCo I'm not 100% sure. I saw this happening in a few builds like [this one](https://app.circleci.com/pipelines/github/salesforcecli/leif/102/workflows/58ac3a5b-d1fb-4071-bffc-c54d2af6bbb2/jobs/101/parallel-runs/0/steps/0-105).

My best guess is something like this
1. Leif runs and creates a sync branch and a PR
2. PR gets squash merged or main is rebased on top of PR branch (or something else weird) *but branch doesn't get deleted*.
3. Leif runs again, sees there's a difference between existing sync branch and main branch (not sure why)
4. Leif tries to create PR but there are no _actual_ file differences if you're trying to merge existing sync branch into main branch.
5. GH API returns a 422 error.

What I ended up doing to "fix" those builds before this PR gets merged was deleted the existing "sync" branches that were sitting around from when Leif had run previously on the repos that this was failing on, then when Leif ran again it created new sync branches based off main and, correctly, didn't find any differences so didn't try to create a PR.

**Comment by RodEsp** [comment]
Whether or not changes are commited is based on `git status` not on `git diff`. So this happens in the scenario where `git status` says there are no new changes but `git diff` says there are.
https://github.com/RasPhilCo/leif/blob/297aaf855826463c4beb665d37fdf310eb71277a/src/asserters/base.ts#L87-L96

**Comment by RasPhilCo** [comment]
@RodEsp great find! i'm gonna aim to fix that now

**Comment by RodEsp** [comment]
@RasPhilCo sounds good! Throw something on my calendar 😄 
And yeah, I'm not 100% sure what the repo steps are here so just making "educated" guesses haha.

**Comment by RasPhilCo** [comment]
@RodEsp Do you mind moving your flag fix to a separate PR for now? I just want to get to the bottom of why the diff isn't doing what's expected before patching it.

**Comment by RodEsp** [comment]
@RasPhilCo just saw this happen again!
https://app.circleci.com/pipelines/github/salesforcecli/leif/128/workflows/9f5db194-7f9d-4fdc-bd26-6b67f2bfba7b/jobs/148?invite=true#step-105-709
Here's the state of that repo when this happened. Red arrow is pointing to `main` when this happened.
image

**Pull request closed without merging**